CSE115A/rideshare-price-function
# Feature/implement function

Problem
=======
As a user, 
I want to see the different prices 
So that I can compare the two 

As a developer, 
I want a tool for getting these prices 
So that I have actual data to display to users

Solution
========
Created a Firebase Function that calls Uber and Lyft's API's that returns data such as `displayName`, which is the type of ride and `price`, which is the price for that tier. 

Steps to Verify:
----------------
1. Checkout out this branch 
1. Within the `/functions` directory, do the following: 
   * Install dependencies within `/functions` directory using `yarn`
   * Run `firebase functions:config:get > .runtimeconfig.json` within the `/functions` directory
1. Run `yarn serve` to host the function locally
1. Open Postman or Insomnia and send a request with any number of the following query params:
```json 
{
   "start_lat": 123,
   "start_lng": 123, 
   "end_lat": 123, 
   "end_lng": 123
}
```
1. Verify that a `200 OK` response is achieved **if and only if** all the parameters are present. 
1. Verify that the lack of query parameters will achieve a `400 Bad Request` Error. 

Screenshots:
-----------------------
**Example of `200 OK` Response** 
<img width="1794" alt="Screen Shot 2020-11-09 at 9 26 20 PM" src="https://user-images.githubusercontent.com/40477553/98631408-39046d00-22d2-11eb-88f2-d53ad19524ee.png">

**Example of `400 Bad Request` Response**
<img width="1794" alt="Screen Shot 2020-11-09 at 9 31 02 PM" src="https://user-images.githubusercontent.com/40477553/98631740-dfe90900-22d2-11eb-9a18-99034bbff583.png">
